frame/support/procedural/src/construct_runtime/parse.rs

```diff
@@ -96,95 +93,18 @@ impl Parse for RuntimeDeclaration {
 			PalletsConversion::Implicit(pallets) =>
 				Ok(RuntimeDeclaration::Implicit(ImplicitRuntimeDeclaration {
 					name,
-					where_section,
 					pallets,
 				})),
 			PalletsConversion::Explicit(pallets) =>
 				Ok(RuntimeDeclaration::Explicit(ExplicitRuntimeDeclaration {
 					name,
-					where_section,
 					pallets,
 					pallets_token,
 				})),
 		}
 	}
 }
 
-#[derive(Debug)]
-pub struct WhereSection {
-	pub block: syn::TypePath,
-	pub node_block: syn::TypePath,
-	pub unchecked_extrinsic: syn::TypePath,
-}
-
-impl Parse for WhereSection {
```

Comment:
We should issue a deprecation warning and not directly let it fail.

https://github.com/paritytech/substrate/blob/03c01e096b4f139ed1b7de35cb07c26371a05674/frame/support/procedural/src/pallet/expand/call.rs#L68-L77 

here is an example on how to produce a deprecated message.